src/hooks/useGoogleAuth.tsx

```diff
@@ -0,0 +1,39 @@
+import { useState } from "react";
+
+const useGoogleAuth = (url: string) => {
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState("");
+  console.log(loading);
+
+  const handleGoogleLogin = async (response: any) => {
+    setLoading(true);
+    fetch(url, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+
+      body: JSON.stringify({ credential: response.credential }),
+    })
+      .then((res) => {
+        setLoading(false);
+
+        return res.json();
+      })
+      .then((data) => {
+        if (data?.user) {
+          localStorage.setItem("user", JSON.stringify(data?.user));
+          window.location.reload();
+        }
+
+        throw new Error(data?.message || data);
+      })
+      .catch((error) => {
+        setError(error?.message);
+      });
+  };
+
+  return { loading, error, handleGoogleLogin };
+};
+
+export default useGoogleAuth;
```

Comment:
Hi, @musllim 

May you use the barrel export way?!

That is, export everything from a single `index.ts` file inside the hooks folder.

Thanks.